### PR TITLE
fix/resources-logic

### PR DESCRIPTION
**Pull request by itsTony4dev**

### CODE REVIEW

File: packages/backend/src/api/resources/resourceService.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ export const resourceService = {
   },
   updateResource: async (
     id: string,
-    updateResourceRequest: PutResourceRequest['body']
+    updateResourceRequest: PutResourceRequest
```
Comment (Hkataya) [inline packages/backend/src/api/resources/resourceService.ts:104]: Body should remain

File: packages/backend/src/api/resources/resourceModel.ts [hunk 2]
```diff
@@ -4,13 +4,20 @@ import { z } from 'zod';
...
+  publish: z.boolean().default(false),
+  tags: z.string(),
+  type: z.string(),
+});
 
+export type SavedResource = z.infer<typeof SavedResourceSchema>;
+export const SavedResourceSchema = z.object({
+  id: z.number(),
+  resourceId: z.string().uuid(),
+  userId: z.string().uuid(),
```
Comment (Hkataya) [inline packages/backend/src/api/resources/resourceModel.ts:21]: shouldnt we use stringId in common validations?

**Review by Hkataya — CHANGES_REQUESTED** [review]

File: packages/backend/src/api/resources/resourceModel.ts (hunk 2 above)
Comment (itsTony4dev) [reply packages/backend/src/api/resources/resourceModel.ts:21]: actually I didn't understand the `stringId` it said that Id must be numeric value and num >0, and a uuid isnt a numeric value

Comment (Hkataya) [reply packages/backend/src/api/resources/resourceModel.ts:21]: ohh you're right i mixed up, keep it like this 👍

**Review by Hkataya — APPROVED** [review]

**Pull request merged**